02_oop/src/mediaplayer/player/playergui.cpp

```diff
@@ -0,0 +1,288 @@
+		connect(mVolumeMute, SIGNAL(clicked()), SLOT(slotSetMute()));
+		connect(mpMediaPlayer, SIGNAL(mediaStatusChanged(QMediaPlayer::MediaStatus)), SLOT(changeMediaStatus(QMediaPlayer::MediaStatus)));
+		// control playlist
+		connect(mpPlaylist, &PlaylistGui::getMediaFileForPlay, this, &PlayerGui::getFileForPlay);
+		connect(mpPlaylist, &PlaylistGui::getMediaFile, this, &PlayerGui::getFile);
+		connect(this, SIGNAL(playCurrentFile()), mpPlaylist, SIGNAL(currentMediaFile()));
+		
+		connect(mSeekForwardButton, SIGNAL(clicked()), mpPlaylist, SIGNAL(nextMediFile()));
+		connect(mSeekBackwordButton, SIGNAL(clicked()), mpPlaylist, SIGNAL(previosMediFile()));
+		// change repeat mode
+		connect(mRepModeButton, SIGNAL(clicked()), SLOT(slotChangeRepMode()));
+		connect(this, SIGNAL(changePlaylistRepeat(bool)), mpPlaylist, SIGNAL(setRepeatAll(bool)));
+	}
+	
+	void PlayerGui::getFile(std::shared_ptr<MediaFile> file) {
+		mCurrPlayFile = file;
+		if (!file)
+			return;
+		auto status = mpMediaPlayer->state();
+		mpMediaPlayer->setMedia(QUrl::fromLocalFile(QString::fromStdString(mCurrPlayFile->path())));
+		if (status != QMediaPlayer::StoppedState)
+			play();
+	}
+	
+	void PlayerGui::getFileForPlay(std::shared_ptr<MediaFile> file) {
+		mCurrPlayFile = file;
+		if (!file)
+			return;
+		play();
+	}
+	
+	void PlayerGui::openPlaylist() {
+		mpPlaylist->isVisible() ? mpPlaylist->hide() : mpPlaylist->show();
+	}
+	
+	void PlayerGui::openLibrary() {
+		mpLibrary->isVisible() ? mpLibrary->hide() : mpLibrary->show();
+	}
+	
+	bool PlayerGui::play() {
+		if (mCurrPlayFile) {
+			mpMediaPlayer->setMedia(QUrl::fromLocalFile(QString::fromStdString(mCurrPlayFile->path())));
+			mpMediaPlayer->play();
+			if (mpMediaPlayer->state() == QMediaPlayer::PlayingState) {
+				mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPause));
+				mpShowMedia->setText(QString::fromStdString(mCurrPlayFile->path()));
+				return true;
+			}
+		}
+		mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+		return false;
+	}
+	
+	void PlayerGui::slotPlay() {
+		switch(mpMediaPlayer->state()) {
+			case QMediaPlayer::PlayingState:
+				mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+				mpMediaPlayer->pause();
+				break;
+			case QMediaPlayer::PausedState:
+				mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPause));
+				mpMediaPlayer->play();
+				break;
+			case QMediaPlayer::StoppedState:
+				if (!play())
+					emit playCurrentFile();
+				break;
+		}
+	}
+
+	void PlayerGui::slotStop() {
+		mpMediaPlayer->stop();
+		mPlayButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+	}
+	
+	void PlayerGui::slotSetMediaPosition(int n) {
+		mpMediaPlayer->setPosition(n);
+	}
+	
+	QString PlayerGui::msecsToString(qint64 sec) {
+		int hours = (sec / (60 * 60 * 1000));
+		int minutes = ((sec % (60 * 60 * 1000)) / (60 * 1000));
+		int seconds = ((sec % (60 * 1000)) / 1000);
+		return QTime(hours, minutes, seconds).toString("hh:mm:ss");
+	}
```

Comment:
This function can be related not only for this class, so can be implemented in the separate class/lib.

Reply:
Function name: **msecs**ToString
Parameter name: **sec**
Where is the truth?

Reply:
Don't invent time transformations. Use existing solutions: `std::chrono::duration_cast` or `QTime::fromMSecsSinceStartOfDay(int msecs)`